src/coreclr/jit/emit.cpp

```diff
@@ -786,21 +789,25 @@ insGroup* emitter::emitSavIG(bool emitAdd)
     // Record how many instructions and bytes of code this group contains
 
     noway_assert((BYTE)emitCurIGinsCnt == emitCurIGinsCnt);
+    noway_assert((BYTE)emitCurIGZeroSzCnt == emitCurIGZeroSzCnt);
     noway_assert((unsigned short)emitCurIGsize == emitCurIGsize);
 
-    ig->igInsCnt = (BYTE)emitCurIGinsCnt;
-    ig->igSize   = (unsigned short)emitCurIGsize;
+    ig->igInsCnt    = (BYTE)emitCurIGinsCnt;
+    ig->igZeroSzCnt = (BYTE)emitCurIGZeroSzCnt;
+    ig->igSize      = (unsigned short)emitCurIGsize;
     emitCurCodeOffset += emitCurIGsize;
     assert(IsCodeAligned(emitCurCodeOffset));
 
 #if EMITTER_STATS
     emitTotalIGicnt += emitCurIGinsCnt;
+    emitTotalIGZeroSzCnt += emitCurIGZeroSzCnt;
     emitTotalIGsize += sz;
     emitSizeMethod += sz;
 
     if (emitIGisInProlog(ig))
     {
         emitCurPrologInsCnt += emitCurIGinsCnt;
+        emitCurPrologZeroSzCnt += emitCurIGZeroSzCnt;
```

Comment:
Do you see us emitting elided moves in the prolog? Seems like we should never have these, both because we should be careful about the code we emit, and because we're not doing any GC tracking there.

Reply:
I didn't check, I just updated it for consistency. Happy to remove this and add an appropriate check so we don't carry these moves in the prolog.

Is there such a check?

Reply:
Also noting that we aren't emitting anything for these `instrDesc`, so if we aren't doing GC updates in the prolog, we shouldn't with this change either. We'd just be unnecessarily creating extra `instrDesc`